### PR TITLE
Update `Math.fround` implementation to a much smaller one.

### PR DESCRIPTION
**Pull request by ljharb**
From https://github.com/paulmillr/es6-shim/pull/140#issuecomment-91787165.

This appears to pass tests across many Firefox/Chrome/IE/Safari/Opera versions.

### CODE REVIEW

**Comment by paulmillr** [comment]
Like it! Less code, same features.

File: es6-shim.js [hunk 1]
```diff
@@ -1362,6 +1224,14 @@
 
   var square = function (n) { return n * n; };
   var add = function (a, b) { return a + b; };
+  var inverseEpsilon = 1 / Number.EPSILON;
+  var roundTiesToEven = function roundTiesToEven(n) {
+    // Even though this reduces down to `return n`, it takes advantage of built-in rounding.
+    return (n + inverseEpsilon) - inverseEpsilon;
+  };
+  var froundEpsilon = Math.pow(2, -23);
```
Comment (Yaffle) [inline es6-shim.js]: better name: BINARY32EPSILON, BINARY32MAX_VALUE and BINARY32MIN_VALUE

File: es6-shim.js [hunk 2]
```diff
@@ -1362,6 +1224,14 @@
 
   var square = function (n) { return n * n; };
   var add = function (a, b) { return a + b; };
+  var inverseEpsilon = 1 / Number.EPSILON;
```
Comment (Yaffle) [inline es6-shim.js]: Why do you need this constant?

File: es6-shim.js [hunk 3]
```diff
@@ -1536,11 +1406,22 @@
     },
 
     fround: function fround(x) {
-      if (x === 0 || x === Infinity || x === -Infinity || Number.isNaN(x)) {
+      if (x === 0 || x === Infinity || x === -Infinity || numberIsNaN(x)) {
         return x;
       }
-      var num = Number(x);
-      return numberConversion.toFloat32(num);
+      var v = Number(x);
```
Comment (Yaffle) [inline es6-shim.js]: convertion to number is done after x === 0 || x === Infinity ... checks...

File: es6-shim.js (hunk 1 above)
Comment (ljharb) [inline es6-shim.js]: much better, thanks

File: es6-shim.js (hunk 2 above)
Comment (ljharb) [inline es6-shim.js]: ¯\_(ツ)_/¯ just avoids recomputing it multiple times.

File: es6-shim.js (hunk 3 above)
Comment (ljharb) [inline es6-shim.js]: https://people.mozilla.org/~jorendorff/es6-draft.html#sec-function-properties-of-the-math-object https://people.mozilla.org/~jorendorff/es6-draft.html#sec-math.acosh - good call, I'll move that line higher.

**Pull request merged**